pkg/controllers/management/globaldns/globaldns_handler.go

```diff
@@ -0,0 +1,228 @@
+package globaldns
+
+import (
+	"context"
+	"net"
+	"sort"
+
+	v1beta1Rancher "github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientv1beta1 "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+)
+
+const (
+	GlobaldnsController = "mgmt-global-dns-controller"
+)
+
+type GlobalDNSController struct {
+	globalDNSInterface v3.GlobalDNSInterface
+	globalDNSLister    v3.GlobalDNSLister
+	//ingressInterface   clientv1beta1.IngressInterface //need to use client-go IngressInterface to update Ingress.Status field
+	ingressLister     v1beta1Rancher.IngressLister
+	ctx               context.Context
+	managementContext *config.ManagementContext
+}
+
+func newGlobalDNSController(ctx context.Context, mgmt *config.ManagementContext) *GlobalDNSController {
+	extInterface, err := v1beta1Rancher.NewForConfig(mgmt.RESTConfig)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	n := &GlobalDNSController{
+		globalDNSInterface: mgmt.Management.GlobalDNSs(""),
+		globalDNSLister:    mgmt.Management.GlobalDNSs("").Controller().Lister(),
+		ctx:                ctx,
+		ingressLister:      extInterface.Ingresses("default").Controller().Lister(),
+		managementContext:  mgmt,
+	}
+	return n
+}
+
+func (n *GlobalDNSController) getIngressInterface(namespace string) clientv1beta1.IngressInterface {
+	if namespace != "" {
+		return n.managementContext.K8sClient.Extensions().Ingresses(namespace)
+	}
+	return nil
+}
+
+func Register(ctx context.Context, management *config.ManagementContext) {
+	n := newGlobalDNSController(ctx, management)
+	management.Management.GlobalDNSs("").AddHandler(ctx, GlobaldnsController, n.sync)
+}
+
+//sync is called periodically and on real updates
+func (n *GlobalDNSController) sync(key string, obj *v3.GlobalDNS) (runtime.Object, error) {
+	logrus.Debugf("GlobalDNSController called")
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	//check if status.endpoints is set, if yes create a dummy ingress if not already present
+	//if ingress exists, update endpoints if different
+
+	if len(obj.Status.Endpoints) > 0 {
```

Comment:
Nit: can we just return right away if `obj.Status.Endpoints == 0`? instead of wrapping the underlying code in if/else block

Reply:
done